### PR TITLE
[Devtools/file upload] Asset

### PR DESCRIPTION
**Pull request by lexfm**
### Change Type (required)
Implemented a file type Input asset version for content uploading to Player data instance.

- [x] `patch`
- [ ] `minor`
- [ ] `major`

### CODE REVIEW

**Comment by mercillo** [comment]
https://github.com/player-ui/devtools-assets/blob/9033f5a6cfbec9d4d98e7ab1f072bccaf0a3bac0/docs/site/pages/assets/input.mdx#L42 can we also update the docs for input that shows support for fileinputs

File: docs/site/pages/assets/input.mdx [hunk 1]
```diff
@@ -6,6 +6,8 @@
 
 It provides an `Input` component that can be used to acquire data from the user.
 
+The file type input is also available for uploading content by simply setting the `file` prop to `true`.
```
Comment (rafbcampos) [inline docs/site/pages/assets/input.mdx:9]: Can we add that the uploaded files will persist in the Player data as strings?

File: input/src/components/hooks.ts [hunk 2]
```diff
@@ -201,3 +201,35 @@ export const useInputAssetProps = (
     value: localValue,
   };
 };
+
+/** Props for file type Input */
+export const useFileInputAssetProps = (
+  props: TransformedInput & { handleFile: (name: string) => void }
+) => {
+  /** Parses file content for upload into a string if file type Inpu */
```
Comment (rafbcampos) [inline input/src/components/hooks.ts:209]:
```suggestion
  /** Parses file content for upload into a string if file type Input */
```

File: input/src/components/hooks.ts [hunk 3]
```diff
@@ -201,3 +201,35 @@ export const useInputAssetProps = (
...
+    if (file) {
+      reader.readAsText(file);
+      props.handleFile(file.name);
+    }
+  };
+
+  return {
+    type: "file",
+    onChange: onFileUpload,
+    accept: ".json, .txt",
```
Comment (rafbcampos) [inline input/src/components/hooks.ts:233]: Should we allow the user to define those? We can default to `.json`, for example, but we could let them decide if they want to accept `.jsx` or `.tsx`.

**Comment by mercillo** [comment]
we may need some tests too since codecov is reporting 0%

**Comment by lexfm** [comment]
> we may need some tests too since codecov is reporting 0%

I still need to understand how `codecov` calculates these parameters, but I believe that 0% seems to be how much more code is test-covered compared to the  target branch. I'm definitely open to discuss about ways or a convention of testing better for new assets, other than only testing the DSL.

**Review by rafbcampos — APPROVED** [review]

**Pull request merged**